db/kv_dense_block.go

```diff
@@ -264,6 +264,42 @@ func (d *DenseBlock) RemoveEntrySet(entries []*LogEntry, bucket base.Bucket) (pe
 	return pendingRemoval, nil
 }
 
+// Removes all entries greater than vb, seq from the block.  Returns rollbackComplete=true if it finds a seq
+// in the block for the vbucket where seq <= rollbackSeq
+func (d *DenseBlock) RollbackTo(rollbackVbNo uint16, rollbackSeq uint64, bucket base.Bucket) (rollbackComplete bool, err error) {
+
+	numRemoved := 0
+	numRemoved, rollbackComplete = d.rollbackEntries(rollbackVbNo, rollbackSeq)
+
+	// If nothing was removed, don't update the block
+	if numRemoved == 0 {
+		return rollbackComplete, nil
+	}
+
+	casOut, writeErr := base.WriteCasRaw(bucket, d.Key, d.value, d.cas, 0, func(value []byte) (updatedValue []byte, err error) {
```

Comment:
I don't disagree, but the current format is consistent with the rest of our WriteCas calls.  Could go through and refactor them all at some point, but don't think it's needed now.